osgp/protocol-adapter-dlms/osgp-protocol-adapter-dlms/src/test/java/org/opensmartgridplatform/adapter/protocol/dlms/infra/messaging/responses/from/core/processors/GetFirmwareFileResponseMessageProcessorTest.java

```diff
@@ -176,7 +177,12 @@ public void processMessageShouldSendNotOkResponseMessageContainingOriginalFirmwa
     // assert
     verify(this.responseMessageSender, times(1)).send(responseMessageArgumentCaptor.capture());
 
-    assertThat(responseMessageArgumentCaptor.getValue().getDataObject()).isSameAs("fw");
+    assertThat(responseMessageArgumentCaptor.getValue().getDataObject())
+        .isInstanceOf(UpdateFirmwareRequestDto.class);
+    assertThat(
+            ((UpdateFirmwareRequestDto) responseMessageArgumentCaptor.getValue().getDataObject())
+                .getFirmwareIdentification())
+        .isSameAs(firmwareFileDto.getFirmwareIdentification());
     assertThat(responseMessageArgumentCaptor.getValue().getResult())
         .isSameAs(ResponseMessageResultType.NOT_OK);
     assertThat(responseMessageArgumentCaptor.getValue().bypassRetry()).isFalse();
```

Comment:
Consider: 
* removing the isInstanceOf check, as the test will still fail on the next assertion when it is not of the correct type
* putting the captured response message in a variable to increase readability

The code may then become something like:
```suggestion
    ResponseMessage capturedValue = responseMessageArgumentCaptor.getValue();
    assertThat(((UpdateFirmwareRequestDto) capturedValue.getDataObject())                .getFirmwareIdentification()).isSameAs(firmwareFileDto.getFirmwareIdentification());
    assertThat(capturedValue.getResult()).isSameAs(ResponseMessageResultType.NOT_OK);
    assertThat(capturedValue.bypassRetry()).isFalse();
```